src/world.rs

```diff
@@ -779,6 +909,37 @@ impl Default for World {
     fn default() -> Self { Self::new() }
 }
 
+/// Describes how to handle a clone_merge. Allows the user to transform components from one type
+/// to another and provide their own implementation for cloning/transforming
+pub trait CloneMergeImpl {
```

Comment:
Can this maybe be default implemented for a T: (Tuple,) type of some sort? I dislike forcing trait implementations (we avoid it everywhere else) when we can standard implement them.

Reply:
Hmm I don't think we can? To support cloning, the caller will need to maintain a registry with function pointers to each type's clone implementation. Components are dynamically discovered during iteration: we don't know which types will exist in legion's storage at the call site.